File: src/main/java/no/novari/flyt/integration/model/entities/Integration.java
package no.novari.flyt.integration.model.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(uniqueConstraints = {
        @UniqueConstraint(
                name = "UniqueSourceApplicationIdAndSourceApplicationIntegrationId",
                columnNames = {"sourceApplicationId", "sourceApplicationIntegrationId"}
        )
})
public class Integration {

    public enum State {
        ACTIVE,
        DEACTIVATED
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private long id;

    @NotNull
    private Long sourceApplicationId;

    @NotBlank
    private String sourceApplicationIntegrationId;

    @NotBlank
    private String destination;

    @NotNull
    @Enumerated(EnumType.STRING)
    private State state;

    private Long activeConfigurationId;

}

File: src/main/kotlin/no/novari/flyt/integration/persistence/entity/Integration.kt
package no.novari.flyt.integration.persistence.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import jakarta.validation.constraints.NotBlank

@Entity
@Table(
    uniqueConstraints = [
        UniqueConstraint(
            name = "UniqueSourceApplicationIdAndSourceApplicationIntegrationId",
            columnNames = ["source_application_id", "source_application_integration_id"],
        ),
    ],
)
class Integration(
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @field:Column(name = "source_application_id", nullable = false)
    var sourceApplicationId: Long? = null,
    @field:NotBlank
    @field:Column(name = "source_application_integration_id")
    var sourceApplicationIntegrationId: String? = null,
    @field:NotBlank
    @field:Column(name = "destination")
    var destination: String? = null,
    @field:Enumerated(EnumType.STRING)
    @field:Column(name = "state", nullable = false)
    var state: State? = State.DEACTIVATED,
    @field:Column(name = "active_configuration_id")
    var activeConfigurationId: Long? = null,
) {
    enum class State {
        ACTIVE,
        DEACTIVATED,
    }
}
